### PR TITLE
Yaka/req size

### PR DESCRIPTION
**Pull request by aliyakamercan**
Removing input size tracking from codec into input controller. Max input size can be configured using with `ControllerConfig`

@DanSimon @nsauro

### CODE REVIEW

File: colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala [hunk 1]
```diff
@@ -60,6 +62,17 @@ class InputControllerSpec extends ColossusSpec with CallbackMatchers{
       //now it should be done
       executed must equal(true)
     }
+    
+    "reject data above the size limit" in {
+      val input = ByteString("hello")
+      val config = ControllerConfig(4, 50.milliseconds, "test-controller", 2L)
+      val con = static(config)
+      intercept[ParseException] {
```
Comment (nsauro) [inline colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala]: Do we want to take this time and change this exception to something more specific. Technically, this is not a parsing issue, its a data size issue.  I would vote we throw a different type of exception specific to the actual cause of too large input.

File: colossus/src/main/scala/colossus/protocols/websocket/Websocket.scala [hunk 2]
```diff
@@ -166,7 +166,8 @@ object FrameParser {
   }
 }
 
-abstract class WebsocketHandler(context: Context) extends Controller(new WebsocketCodec, ControllerConfig(50, scala.concurrent.duration.Duration.Inf), context) {
+abstract class WebsocketHandler(context: Context)
```
Comment (nsauro) [inline colossus/src/main/scala/colossus/protocols/websocket/Websocket.scala]: Shouldn't this be taking in a configuration, and then passing that to the parent?

File: colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala (hunk 1 above)
Comment (DanSimon) [inline colossus-tests/src/test/scala/colossus/controller/InputControllerSpec.scala]: I think you're right but it probably shouldn't be done here.  Right now users can't catch these exceptions anyway do it doesn't really matter what the type is.  #263 needs to happen

File: colossus/src/main/scala/colossus/protocols/websocket/Websocket.scala (hunk 2 above)
Comment (DanSimon) [inline colossus/src/main/scala/colossus/protocols/websocket/Websocket.scala]: I agree, but it's not really relevant to this PR, I'll create an issue for this.

**Comment by DanSimon** [comment]
:+1:  the code coverage decrease seems to just be in constructors so I'm not too worried about that.

**Comment by nsauro** [comment]
:pizza:

**Pull request merged**